core/src/main/java/com/linecorp/armeria/server/ArmeriaHttpException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.HttpStatus;
+
+import io.netty.handler.codec.http2.Http2Error;
+
+/**
+ * A {@link RuntimeException} that is raised when an armeria internal http exception has occurred.
+ * This class is the general class of exceptions produced by a failed request or a reset stream.
+ */
+public abstract class ArmeriaHttpException extends RuntimeException {
```

Comment:
- A user may want to instantiate this exception?
- How about overriding `fillInStackTrace()` and doing nothing when `Flags.verboseExceptions()` is disabled?
- Could be renamed so it doesn't have `Armeria` in its name? e.g. `HttpResponseException`